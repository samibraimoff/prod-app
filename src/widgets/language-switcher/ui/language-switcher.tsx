import { useTranslation } from 'react-i18next';
import { classNames } from 'shared/lib';
import { Button, ButtonTheme } from 'shared/ui';

import classes from './language-switcher.module.scss';

interface LanguageSwitcherProps {
  className?: string;
}

export function LanguageSwitcher(props: LanguageSwitcherProps) {
  const { className } = props;
  const { t, i18n } = useTranslation();
  const changeLang = () => (i18n.language === 'en-US'
    ? i18n.changeLanguage('ru-RU')
    : i18n.changeLanguage('en-US'));

  return (
    <Button
      data-testid="language-switcher"
      theme={ButtonTheme.CLEAR}
      onClick={changeLang}
      className={classNames(classes['language-switcher'], {}, [className])}
    >
      {t('language')}
    </Button>
  );
}
