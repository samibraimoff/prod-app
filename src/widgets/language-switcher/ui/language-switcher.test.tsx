import { screen, render } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import { axe } from 'jest-axe';
import { useTranslation } from 'react-i18next';
import { LanguageSwitcher } from './language-switcher';

jest.mock('react-i18next');
const mockChangeLanguage = jest.fn();

(useTranslation as jest.Mock).mockReturnValue({
  t: (key: string) => key,
  i18n: { language: 'en-US', changeLanguage: mockChangeLanguage },
});

describe('LanguageSwitcher', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should change language', async () => {
    render(<LanguageSwitcher />);
    const button = screen.getByTestId('language-switcher');
    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent('language');
    await userEvent.click(button);
    expect(mockChangeLanguage).toBeCalledTimes(1);
    expect(mockChangeLanguage).toBeCalledWith('ru-RU');
  });

  it('should not have any a11y violations', async () => {
    const { container } = render(<LanguageSwitcher />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should match snapshot', () => {
    const { container } = render(<LanguageSwitcher />);
    expect(container).toMatchSnapshot();
  });

  it('should apply theme and class name', () => {
    render(<LanguageSwitcher className="test" />);
    expect(screen.getByTestId('language-switcher')).toHaveClass('test');
  });
});
