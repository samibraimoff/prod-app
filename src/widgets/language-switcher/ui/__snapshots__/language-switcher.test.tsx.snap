// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSwitcher should match snapshot 1`] = `
<div>
  <button
    class="button language-switcher clear"
    data-testid="language-switcher"
    type="button"
  >
    language
  </button>
</div>
`;
